### PR TITLE
Admin setup compilers and core libraries

### PR DESCRIPTION
**Pull request by psakievich**
Create a way to bootstrap the same compilers, mpi etc on a given system. Right now targeting `cee` and `snl-hpc`.

### CODE REVIEW

File: scripts/bootstrap_tpls/pcomps/spack.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  specs:
+  - gcc@10.3.0
+  - llvm@13+clang targets=x86 ~gold ^ninja@kitware
+  view: false
+  concretizer:
+    unify: false
+  packages:
+    all:
+      target: [x86_64]
+      variants: [build_type=Release]
```
Comment (jrood-nrel) [inline scripts/bootstrap_tpls/pcomps/spack.yaml]: I think you will want to use `requires` in these so you don't get unexpected behavior.

File: scripts/bootstrap_tpls/Makefile [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+export SPACK_USER_CACHE_PATH=$(CURDIR)/.spack
+
+.SUFFIXES:
+
+.PHONY: all clean
+
+all: tpls/generated/env
+
+clean:
+	rm -rf icomp/spack.mk icomp/spack.lock icomp/compilers.yaml icomp/config.yaml icomp/packages.yaml icomp/generated icomp/.spack-env
```
Comment (jrood-nrel) [inline scripts/bootstrap_tpls/Makefile]: I think the `-r` can be removed here for safety.

Comment (jrood-nrel) [reply scripts/bootstrap_tpls/Makefile]: Maybe there are directories in there I guess.

Comment (psakievich) [reply scripts/bootstrap_tpls/Makefile]: Yeah `.spack-env` is a directory

File: scripts/bootstrap_tpls/pcomps/spack.yaml (hunk 1 above)
Comment (psakievich) [reply scripts/bootstrap_tpls/pcomps/spack.yaml]: For this case it _shouldn't_ matter. Each install tree is blank so there is nothing for the concretizer to choose over the suggestions we give it. But yeah `requires` is probably safer.

**Review by eugeneswalker — APPROVED** [review]

**Review by jrood-nrel — APPROVED** [review]

**Pull request merged**